### PR TITLE
[MISC] Separate kmer counting from sketches.

### PR DESCRIPTION
**Pull request by smehringer**
This makes everything easier once we can read sketches from a file.

### CODE REVIEW

File: include/hibf/sketch/compute_sketches.hpp [hunk 1]
```diff
@@ -17,12 +17,9 @@ namespace seqan::hibf::sketch
 /*!\brief Computes the kmer_counts and sketches and stores them in the respective vectors for further use.
  * \ingroup hibf_layout
  * \param[in] config The configuration to compute the layout with.
- * \param[in,out] kmer_counts The vector that will store the kmer counts (estimations).
  * \param[in,out] hll_sketches The vector that will store the sketches.
  */
-void compute_sketches(config const & config,
-                      std::vector<size_t> & kmer_counts,
-                      std::vector<sketch::hyperloglog> & hll_sketches);
+void compute_sketches(config const & config, std::vector<sketch::hyperloglog> & hll_sketches);
```
Comment (eseiler) [inline include/hibf/sketch/compute_sketches.hpp:22]: What about

```cpp
void compute_sketches(config const & config,
                      std::vector<size_t> & kmer_counts,
                      std::vector<sketch::hyperloglog> & hll_sketches,
                      std::vector<sketch::minhashes> & minhash_sketches);
```

?

File: src/hierarchical_interleaved_bloom_filter.cpp [hunk 2]
```diff
@@ -204,10 +205,12 @@ hierarchical_interleaved_bloom_filter::hierarchical_interleaved_bloom_filter(con
 {
     configuration.validate_and_set_defaults();
 
-    std::vector<size_t> kmer_counts{};
     std::vector<sketch::hyperloglog> sketches{};
     layout_compute_sketches_timer.start();
-    sketch::compute_sketches(configuration, kmer_counts, sketches);
+    sketch::compute_sketches(configuration, sketches);
+    std::vector<size_t> kmer_counts{};
+    kmer_counts.resize(configuration.number_of_user_bins);
```
Comment (eseiler) [inline src/hierarchical_interleaved_bloom_filter.cpp:212]: `hibf::sketch::estimate_kmer_counts` already resizes

File: test/unit/hibf/layout/compute_layout_test.cpp [hunk 3]
```diff
@@ -33,7 +34,9 @@ TEST(compute_layout, dispatch)
     std::vector<seqan::hibf::sketch::hyperloglog> sketches;
     std::vector<size_t> kmer_counts;
 
-    seqan::hibf::sketch::compute_sketches(config, kmer_counts, sketches);
+    seqan::hibf::sketch::compute_sketches(config, sketches);
+    kmer_counts.resize(sketches.size());
```
Comment (eseiler) [inline test/unit/hibf/layout/compute_layout_test.cpp:38]: `hibf::sketch::estimate_kmer_counts` already resizes

**Review by eseiler — CHANGES_REQUESTED** [review]

File: include/hibf/sketch/compute_sketches.hpp (hunk 1 above)
Comment (smehringer) [reply include/hibf/sketch/compute_sketches.hpp:22]: I don't even know if I use this overload at all :D. I'll check

Comment (smehringer) [reply include/hibf/sketch/compute_sketches.hpp:22]: Ok I am using it but for consistency it should also be removed.

**Review by eseiler — APPROVED** [review]

**Pull request merged**